model/inline/src/main/scala/aqua/model/inline/raw/ApplyStreamMapRawInliner.scala

```diff
@@ -0,0 +1,367 @@
+package aqua.model.inline.raw
+
+import aqua.errors.Errors.internalError
+import aqua.model.inline.Inline
+import aqua.model.inline.RawValueInliner.unfold
+import aqua.model.inline.state.*
+import aqua.model.{SeqModel, *}
+import aqua.raw.value.IntoArrowRaw
+import aqua.types.StreamMapType.Func.*
+import aqua.types.{
+  ArrayType,
+  CanonStreamMapType,
+  CanonStreamType,
+  DataType,
+  ScalarType,
+  StreamMapType,
+  StreamType,
+  StructType
+}
+
+import cats.data.NonEmptyMap
+import cats.data.{Chain, State}
+import cats.syntax.applicative.*
+import cats.syntax.functor.*
+import cats.syntax.monoid.*
+import cats.syntax.option.*
+
+object ApplyStreamMapRawInliner {
+
+  // make unique name that cannot be used in Aqua
+  private def mang[S: Mangler](name: String, suffix: String): State[S, String] =
+    Mangler[S].findAndForbidName(s"-${name}_$suffix-")
+
+  // - fold a map in parallel
+  // - when a key matches index, push result to a stream
+  // - return the stream
+  private def getStreamModel(
+    mapName: String,
+    mapType: StreamMapType,
+    idxVar: ValueModel,
+    streamVar: VarModel,
+    iterName: String
+  ): OpModel.Tree = {
+    val mapVar = VarModel(mapName, mapType)
+    val arrayResultType = ArrayType(mapType.element)
+    val iter = VarModel(iterName, mapType.iterType("iterName_type"))
+
+    ParModel.wrap(
+      ForModel(iter.name, mapVar, ForModel.Mode.Never).wrap(
+        XorModel.wrap(
+          MatchMismatchModel(
+            iter
+              .withProperty(
```

Comment:
Why not `intoField` method?

Reply:
because it returns option

Reply:
Personally I think `.intoField("key").getOrElse(internalError(...))` is better because than we have single source of truth for how field is called and what is the type.